contrib/NeonVM/neonvm.c

```diff
@@ -298,20 +364,23 @@ void exec_PUSHB()
     exec_error("not implemented");
 }
 
-void exec_PUSHN()
+void exec_PUSHN(struct tagTExecutor *self)
 {
-    exec_error("not implemented");
+    uint32_t val = exec_getOperand(self);
+    push(self->stack, cell_fromNumber(bid128_from_string(self->object->strtable[val])));
 }
 
 void exec_PUSHS(struct tagTExecutor *self)
 {
     uint32_t val = exec_getOperand(self);
-    push(self->stack, self->object->strtable[val]);
+    push(self->stack, cell_fromString(self->object->strtable[val]));
 }
 
-void exec_PUSHPG()
+void exec_PUSHPG(struct tagTExecutor *self)
 {
-    fatal_error("not implemented");
+    uint32_t addr = exec_getOperand(self);
+    assert(addr < self->object->global_size);
+    push(self->stack, cell_fromAddress(&self->globals[addr]));

```

Comment:
Do you want to push the pointer to the cell? Or a pointer to the pointer to the cell?